packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -214,29 +288,56 @@ handler.checkIfPRIsInvalid = async function checkIfPRIsInvalid(
 };
 
 /**
- * Adds a PR to datastore when an automerge label is added to a PR
+ * Adds a PR to datastore; first, checks if there's branch protection and adds a reaction Id
  * @param url type string
  * @param wp type Watch PR (owner, repo, pr number, state, url)
+ * @param github type githup API surface from payload
  * @returns void
  */
-handler.addPR = async function addPR(wp: WatchPR, url: string) {
-  const key = datastore.key([TABLE, url]);
-  const entity = {
-    key,
-    data: {
-      created: new Date().toJSON(),
-      owner: wp.owner,
-      repo: wp.repo,
-      number: wp.number,
-      branchProtection: wp.branchProtection,
-      label: wp.label,
-      author: wp.author,
-      reactionId: wp.reactionId,
-      installationId: wp.installationId,
-    },
-    method: 'upsert',
-  };
-  await datastore.save(entity);
+handler.addPR = async function addPR(
+  wp: WatchPR,
+  url: string,
+  github: Context['github']
```

Comment:
Other bots are using this type: https://github.com/googleapis/repo-automation-bots/blob/78950c856ed7205341efd8908b4cca935ab1c5ab/packages/gcf-utils/src/gcf-utils.ts#L41